e2e/test-environment-circus/testEnvironment.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+module.exports = process.env.ASYNC_HANDLE_TEST_EVENT
```

Comment:
same here, I think just a completely separate test is cleaner than this conditional env

Reply:
Resolved via moving the logic to a separate e2e suite, as recommended.